com.playeveryware.eos/Runtime/Core/EOSCommandLineOverrides.cs

```diff
@@ -0,0 +1,91 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using PlayEveryWare.EpicOnlineServices;
+
+namespace PlayEveryWare.EpicOnlineServices
+{
+    /// <summary>
+    /// Holds override values for Deployment and Sandbox IDs.
+    /// </summary>
+    public static class EOSOverrideState
+    {
+        public static string DeploymentIdOverride { get; set; }
+        public static string SandboxIdOverride { get; set; }
+    }
+}
+/// <summary>
+/// Applies CLI overrides for Epic Online Services before any scene loads.
+/// </summary>
+public static class EOSCommandLineOverrides
+{
+    private const string DeploymentKey = "-epicdeploymentid";
+    private const string SandboxKey = "-epicsandboxid";
+
+    [RuntimeInitializeOnLoadMethod(RuntimeInitializeLoadType.SubsystemRegistration)]
+    private static void ApplyCliOverrides()
+    {
+        try
+        {
+            Debug.Log($"[{nameof(EOSCommandLineOverrides)}::{nameof(ApplyCliOverrides)}] Initializing CLI overrides�");
+
+            // Get command line arguments
+            string[] args = Environment.GetCommandLineArgs();
+            string deploymentArg = null;
+            string sandboxArg = null;
+
+            // Parse CLI arguments for deployment and sandbox IDs
+            foreach (var arg in args)
+            {
+                if (arg.StartsWith(DeploymentKey, StringComparison.OrdinalIgnoreCase))
+                    deploymentArg = arg.Substring(DeploymentKey.Length);
+
+                if (arg.StartsWith(SandboxKey, StringComparison.OrdinalIgnoreCase))
+                    sandboxArg = arg.Substring(SandboxKey.Length);
+            }
+
+            // If no overrides found, exit early
+            if (deploymentArg == null && sandboxArg == null)
```

Comment:
Adding `IsNullOrEmpty()` adds a little extra safety
```suggestion
            if (string.IsNullOrEmpty(deploymentArg) && string.IsNullOrEmpty(sandboxArg))
```

Reply:
Comment applied